### PR TITLE
Run e2e's as standalone

### PR DESCRIPTION
**Pull request by ozdanborne**
Running `ginkgo` with just this file will execute the kubernetes e2e's.

This should enable us to write our own tests and use the Kubernetes testing framework without having to drag all their source files around with us (i.e. fork kubernetes)

Note, this is one layer down than preferred. We are using ginkgo directly instead of the binary that usually calls it (e2e.test? kubetest??). This limits us to just ginkgo flags instead of the normal e2e.test flags, and might have some other sideffects.

### CODE REVIEW

File: mye2e_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	_ "k8s.io/kubernetes/test/e2e/apimachinery"
+	_ "k8s.io/kubernetes/test/e2e/apps"
+	_ "k8s.io/kubernetes/test/e2e/auth"
+	_ "k8s.io/kubernetes/test/e2e/autoscaling"
+	_ "k8s.io/kubernetes/test/e2e/common"
+	_ "k8s.io/kubernetes/test/e2e/instrumentation"
+	_ "k8s.io/kubernetes/test/e2e/kubectl"
+	_ "k8s.io/kubernetes/test/e2e/lifecycle"
+	_ "k8s.io/kubernetes/test/e2e/lifecycle/bootstrap"
+	_ "k8s.io/kubernetes/test/e2e/network"
```
Comment (gunjan5) [inline mye2e_suite_test.go]: can't you just import this one and leave out the rest if we only care about networking e2es?

**Comment by nelljerram** [comment]
Looks good, but to check my understanding here:

- With this, I think we'd then use glide.yaml / .lock files to specify and track precisely what level of the k8s test source we are using.

- We still need a fork (as at https://github.com/tigera/kubernetes) to hold any non-public test code changes that we need (as we do for essentials).  (And we can use glide.yaml to redirect to that fork, of course.)

Right?

**Comment by ozdanborne** [comment]
@neiljerram 

>With this, I think we'd then use glide.yaml / .lock files to specify and track precisely what level of the k8s test source we are using.

Yes.

>We still need a fork (as at https://github.com/tigera/kubernetes) to hold any non-public test code changes that we need (as we do for essentials). (And we can use glide.yaml to redirect to that fork, of course.)

Yes. I wasn't aware we had non-public test code changes. Can you point me at them?

**Comment by nelljerram** [comment]
@ozdanborne https://github.com/tigera/kubernetes/commits/essentials-testing

**Pull request merged**